nibabel/minc1.py

```diff
@@ -319,6 +319,49 @@ def from_file_map(klass, file_map):
             data = klass.ImageArrayProxy(minc_file)
         return klass(data, affine, header, extra=None, file_map=file_map)
 
+    @property
+    def _spatial_dims(self):
+        if len(self.shape) > 3:
+            return slice(1, 4)
+        return slice(0, 3)
+
+    def _check_slicing(self, slicer, return_spatial=False):
+        ''' Canonicalize slicers and check for scalar indices in spatial dims
+
+        Parameters
+        ----------
+        slicer : object
+            something that can be used to slice an array as in
+            ``arr[sliceobj]``
+        return_spatial : bool
+            return only slices along spatial dimensions (x, y, z)
+
+        Returns
+        -------
+        slicer : object
+            Validated slicer object that will slice image's `dataobj`
+            without collapsing spatial dimensions
+        '''
+        slicer = canonical_slicers(slicer, self.shape)
+        try:
+            all_slices = super(Minc1Image, self)._check_slicing(slicer, False)
+            sp_dims = self._spatial_dims
+        except IndexError:
```

Comment:
This is where all my pain is - the problem of which axis is which.   I think we need a better solution, like axis names : https://github.com/nipy/nibabel/wiki/BIAP6 ...

Reply:
Okay. So I've seen BIAP6 come up in one form or another a number of times over the last 5-6 years. What would it take to make a final decision on how to handle axes (mostly: who do we need input from?) and do you have an estimate of the effort involved to make a change here? I'm happy to contribute some effort in hacking and review, if there's a reasonably defined scope.

In the short term, does it make sense to simply raise `NotImplementedError` in MINC, and punt on this issue entirely, as far as this PR is concerned? Or do you think working out the spatial dimensions (as suggested in https://github.com/nipy/nibabel/pull/550/files#r169142529) is small enough to make handling this in MINC feasible?

I recognize that it's dissatisfying to keep adding features that aren't available in MINC, but even if it's only available in NIFTI, this feature seems valuable (and having it in NIFTI might encourage users more familiar with MINC to help extend the functionality).

Reply:
Yes, fair point (about the 5-6 years).   How about a developer hangout soonish to see if we can scope this out?   If we find a proper fix is not practical, we can drop MINC, that's what I did in practice in the [processing module](https://github.com/nipy/nibabel/blob/master/nibabel/processing.py#L158), using the [spatial_axes_first function](https://github.com/nipy/nibabel/blob/master/nibabel/imageclasses.py#L120).

Reply:
That sounds reasonable to me. I'm pretty available between 9am-4pm EST for the next three weeks (also other times, but harder to describe briefly, and less predictable). I'm not sure who else should be involved, but assume you have a better sense of the field of relevant/interested parties? I have an admittedly basic grasp on MINC (I found the standard unsatisfyingly vague, though that may be because it depends on familiarity with HDF5), so that might be worth bearing in mind in terms of the expertise you want involved in the hangout.

Reply:
For now - do you think you can re-use the `spatial_axes_first` function etc from `processing`?  And therefore, disallow MINC?

Reply:
Sure thing.